src/main/scala/is/hail/variant/VariantView.scala

```diff
@@ -0,0 +1,56 @@
+package is.hail.variant
+
+import is.hail.annotations._
+import is.hail.expr._
+
+class VariantView(t: TStruct) {
+  private var region: MemoryBuffer = _
+  private var offset: Long = _
+  private val contigIdx: Int = t.fieldIdx("contig")
+  private val startIdx: Int = t.fieldIdx("start")
+  private val refIdx: Int = t.fieldIdx("ref")
+  private val altAllelesIdx: Int = t.fieldIdx("altAlleles")
+  private val tAltAlleles: TArray = t.fieldType(altAllelesIdx).asInstanceOf[TArray]
+  private val tAltAllele: TStruct = tAltAlleles.elementType.asInstanceOf[TStruct]
+  private val altIdx: Int = tAltAllele.fieldIdx("alt")
+
+  assert(t.fieldType(contigIdx).required)
+  assert(t.fieldType(startIdx).required)
+  assert(t.fieldType(refIdx).required)
+  assert(tAltAlleles.required)
+  assert(tAltAlleles.elementType.required)
+  assert(tAltAllele.fieldType(altIdx).required)
+
+  def setRegion(rv: RegionValue) {
+    region = rv.region
+    offset = rv.offset
+  }
+
+  def setRegion(region: MemoryBuffer, offset: Long) {
+    this.region = region
+    this.offset = offset
+  }
+
+  def getContig(): String = {
+    TString.loadString(region, t.loadField(region, offset, contigIdx))
+  }
+
+  def getStart(): Int = {
+    region.loadInt(t.loadField(region, offset, startIdx))
+  }
+
+  def getRef(): String = {
+    TString.loadString(region, t.loadField(region, offset, refIdx))
+  }
+
+  def getAltAlleles(): Long = {
```

Comment:
Hmm, this breaks the conceptual categories for me.  Although I'm not necessarily against it, I thought views should return Java types (generally primitive to avoid boxing, although String is an exception until we get a better plan for strings.)

Reply:
I agree. Creation of a Java `Array` seems untenable because it recursively forces reification.

The current scheme uses the set of "intermediate values" that the IR manipulates on the stack / in locals. In particular, any non-primitive is `Long`.

The only other reasonable approach seems to be `def getAltAlleles(): ArrayView`.

Reply:
Are you deleting this?

Reply:
Yeah, I need to back port some things from alt allele view into this before this is merged. I'll try to get to it late afternoon, wanna focus on hail lab meeting now.